dvc/output.py

```diff
@@ -747,7 +754,22 @@ def _commit_granular_dir(self, filter_info) -> Optional["HashFile"]:
     def dumpd(self, **kwargs):  # noqa: C901
         meta = self.meta.to_dict()
         meta.pop("isdir", None)
-        ret: Dict[str, Any] = {**self.hash_info.to_dict(), **meta}
+        ret: Dict[str, Any] = {}
+        if (
+            (not self.IS_DEPENDENCY or self.stage.is_import)
+            and self.hash_info.isdir
+            and (kwargs.get("with_files") or self.files is not None)
+        ):
```

Comment:
Any reason to move this upward @daavoo? This might affect the ordering of metadata in `.dvc` file.

Reply:
No real reason. Could be moved bellow

Reply:
Created https://github.com/iterative/dvc/pull/8879.